### ISSUE TITLE
Ensure projects exist when creating and updating teams and tokens

### ISSUE THREAD

**Issue by bcmdarroch** [chef/automate#833]
## User Story

As an Automate User, I will use this work when I want to create/update teams and tokens with projects.

## Definition of Done

On create and update, the Teams and Tokens APIs verify the existence of any projects in the request. 

## Demo Script / Repro Steps

1. `curl -kH "api-token: $TOK" -d '{"id": "new-team", "name": "brand new team", "projects":["not-found"]}' https://a2-dev.test/apis/iam/v2beta/teams | jq`
get an error that project "not-found" does not exist, create fails

2. `curl -kH "api-token: $TOK" -d '{"id": "some-token", "name": "updated name", "projects":["not-found"]}' https://a2-dev.test/apis/iam/v2beta/tokens/some-token | jq`
get an error that project "not-found" does not exist, update fails

## Subtasks

- [ ] Teams
- [ ] Tokens

**Comment by sdelano** [comment]
@blakestier is going to take some time while this is still a ways out to work through some design trade-offs with this story. What was discussed at the team meeting this morning was the trade-off between:
1. the time and effort required to merge the various authz-related services into one
1. the need to solve the general problem of checking existence of projects for non-auth services, e.g. `ingest-service`
1. the potential cyclical dependencies of `teams-service` and `authz-service` making calls to each other and how that type of topology is not supported in our deployments

Additionally, it came up in the team channel that related to this (creating / updating projects on other resources)  how we do authorization on sub-resources. That was documented in the team Slack as:

> # Authorization for Sub Resources
> 
> tl;dr: The authz system will provide basic authz at the gateway based on the user, resource, and action. The domain services are responsible for internal business logic and asking authz questions that are based on any domain-specific business logic.
> 
> ## GETs: Sub-Resource Filtering
>
> The authorization system and the domain services will not do sub-resource filtering based on sub-resource authz rules. If a resource has sub-resources that belong to, or are associated with it, then the identifiers for these resources will be visible in the API responses for the parent resource.
>
> Examples of this are:
> - a token belonging to multiple projects, some of which the current user may not have access to
> - a node manager having multiple node credentials, some of which the current user may not have access to
> 
> This design is a tradeoff between being leaky with identifiers, or building complex APIs and backend logic to infer what a user has changed about a resource when we don't provide them with the entire resource to begin with. We've decided to expose the identifiers to begin with, and as we learn more about the system we can implement solutions to tighten this up as necessary.
> 
> ## POSTs: Sub-Resource Changes
> 
> Domain services are responsible for asking the authz system business-logic authorization questions based on changes to their resources that only they can deduce.
> 
> For example, if a user was to update a node manager to add it to project stephens-team and also add the node credentials victorias-ssh-keys to it, the domain service responsible for node managers would need to ask the authorization system whether:
> 
> - the user is able to assign resources to stephens-team
> - the user is able to modify victorias-ssh-keys
> 
> If either of these cases is false, it is the responsiblity of the domain service to return an authorization error to the user.

**Comment by bcmdarroch** [comment]
I'm going to split these stories in two: this issue will focus on adding project existence verification to Teams/Tokens APIs. The second issue will focus on adding project assignment permission checking to IAM APIs.
